conpype/test/test_task.py

```diff
@@ -0,0 +1,35 @@
+from conpype import Pipeline, PutStep, GetStep, DoStep, GitRepo, Task
+import unittest
+
+
+class TestJobSimple(unittest.TestCase):
+
+    def test_basic(self):
+        with Pipeline("test", script_dirs={"fake": "fake_scripts"}) as pipeline:
+            job = pipeline.job("job")
+
+            with job as job:
+                @job.task()
+                def test_task():
+                    return 0
+
+            obj = job.concourse()
+
+            obj["plan"] = obj["plan"][1:]  # remove init task. it is checked test_pipeline.py
+            self.assertEqual(obj["plan"][0]["task"], "test-task")
+
+    def test_output(self):
+        with Pipeline("test", script_dirs={"fake": "fake_scripts"}) as pipeline:
+            job = pipeline.job("job")
+
+            with job as job:
+                @job.task(outputs=["out"])
+                def test_task(out):
+                    return 0
+
+            obj = job.concourse()
+
+            self.assertEqual(obj["plan"][1]["task"], "test-task")
+            self.assertEqual(obj["plan"][1]["config"]["outputs"][1], {"name": "out"})
```

Comment:
I think this test can be simplified by removing the dependency on `Pipeline` and `Job` entirely: https://github.com/SAP/pipeline-dsl/commit/34ddb9b42fa4761c89784ce4803205458eeaef3c

Unfortunately, a lot of the defaulting for `Task` (e.g. `timeout`, `output`) happens in `Job.task` (and is even duplicated to `ParallelStep.task`), which inflates the test cases. This could be moved like so: https://github.com/SAP/pipeline-dsl/commit/549f7db4e466e7f1ae3083471f89312a614987b1

If you agree, please cherry-pick this commits to your branch :)

Reply:
Done